R/derive_vars_dy.R

```diff
@@ -0,0 +1,107 @@
+#' Derive Relative Day Variables
+#'
+#' Adds relative day variables (`--DY`) to the dataset, e.g., `ASTDY` and
+#' `AENDY`.
+#'
+#' @param dataset Input dataset
+#'
+#'   The columns specified by the `reference_date` and the `source_vars` parameter are
+#'   expected.
+#'
+#' @param reference_date The start date column, e.g., date of first treatment
+#'
+#'   A date or date-time object column is expected.
+#'
+#' @param source_vars A list of datetime or date variables created using `vars()` from
+#'   which dates are to be extracted. This can either be a list of date(time) variables
+#'   or named `--DY` variables and corresponding --DT(M) variables e.g.
+#'   vars(TRTSDTM, ASTDTM, AENDT) or vars(TRTSDT, ASTDTM, AENDT,DEATHDY=DTHDT).
+#'   If the source variable does not end in --DT(M), a name for the resulting
+#'   `--DY` variable must be provided.
+#'
+#' @author Teckla Akinyi
+#'
+#' @details The relative day is derived as number of days from the reference date
+#'   to the end date. If it is nonnegative, one is added. I.e., the relative day
+#'   of the reference date is 1. Unless a name is explicitly specified, the name
+#'   of the resulting relative day variable is generated from the source variable
+#'   name by replacing DT (or DTM as appropriate) with DY.
+#'
+#' @return The input dataset with `--DY` corresponding to the `--DTM` or `--DT`
+#'     source variable(s) added
+#'
+#' @keywords derivation ADaM timing
+#'
+#' @export
+#'
+#' @examples
+#' library(lubridate)
+#' library(dplyr)
+#'
+#' datain <- tibble::tribble(
+#'   ~TRTSDTM, ~ASTDTM, ~AENDT,
+#'  "2014-01-17T23:59:59", "2014-01-18T13:09:O9", "2014-01-20"
+#' ) %>%
+#'  mutate(TRTSDTM = lubridate::as_datetime(TRTSDTM),
+#'         ASTDTM = lubridate::as_datetime(ASTDTM),
+#'         AENDT = lubridate::ymd(AENDT))
+#'
+#' derive_vars_dy(datain, reference_date = TRTSDTM, source_vars = vars(TRTSDTM, ASTDTM, AENDT))
+#'
+#'  datain <- tibble::tribble(
+#'   ~TRTSDT, ~ASTDTM, ~AENDT, ~DTHDT,
+#'  "2014-01-17", "2014-01-18T13:09:O9", "2014-01-20", "2014-02-01"
+#'  ) %>%
+#'  mutate(TRTSDT = lubridate::ymd(TRTSDT),
+#'         ASTDTM = lubridate::as_datetime(ASTDTM),
+#'         AENDT = lubridate::ymd(AENDT),
+#'         DTHDT = lubridate::ymd(DTHDT))
```

Comment:
Are the `lubridate::` necessary if the `library(lubridate)` is run??

Reply:
Nope. Quite distracting in fact.